sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/EliminateUnnecessaryJoin.scala

```diff
@@ -59,19 +66,33 @@ object EliminateUnnecessaryJoin extends Rule[LogicalPlan] {
         case Some(count) => hasRow == (count > 0)
         case _ => false
       }
+
+    case LocalRelation(_, data, isStreaming) if !isStreaming =>
```

Comment:
if we want to handle `LocalRelation`, then it's not AQE specific and we can do it in the normal optimizer?

Reply:
yes, but currently we have no chance to do normal optimizer at AQE side. Maybe we can let some rules which in `Optimizer` also available at `AQEOptimizer` in future ?

Reply:
Since AQE is on by default, it's not a big issue but more about code cleanness. How about this:
1. `EliminateUnnecessaryJoin` should only deal with `LocalRelation`, and appears in both the normal optimizer and AQE optimizer
2. AQE optimizer adds a new rule to turn empty query stage into empty `LocalRelation`

Reply:
sgtm, and just find a exists rule `PropagateEmptyRelation`. 

How about this updating ?
* Add a rule `ConvertToLocalRelation` to  turn empty query stage into empty LocalRelation
* Make `PropagateEmptyRelation` appears in AQE optimizer
* Reduce `EliminateUnnecessaryJoin` to only handle naaj/semi/anti joins